osmlab/to-fix-backend
# #519 Bug for Show Tag Source

# Ticket Number
https://github.com/mapbox/to-fix-internal/issues/519

# Description
Updates tag model to support the name of the user who adds a tag to a particular item.

# Areas of Change
* `database/index.js` : updated association between Tag and Item to be done through the ItemTag table
* `database/itemtag.js` : create the itemtag model.
* `migrations/20180322155410-add_itemtag_table.js` :  migration for updating the db with the new itemtag model
* `routes/tags.js` : modify createItemTag method to fill in `authorName` and `authorId` fields 
* `test/` : update all tests to support new property. 

# Test
- Pull branch bug/tf-519_incorrec-tag-source
- Startup a local instance of Postgres
- Run npm start
- Startup a local instance of to-fix internal and pull in branch bug/tf-519_incorrec-tag-source
- Go to an item
- Add a tag
- Hover over the item to see user name and date added.
- Repeat again if needed.
